src/rust/ide/lib/view/src/graph_editor/component/project_name.rs

```diff
@@ -212,6 +212,10 @@ impl ProjectNameModel {
         let project_name = self.clone_ref();
         //FIXME[dg]: This section to check newline and keep TextField in a single line is hacky
         // and should be removed once the new TextField is implemented.
+        // Also: Taking TextField reference inside a closure managed by itself is a reference
+        // loop, which means we are leaking memory. This cannot be fixed without refactoring
+        // TextField. This code will not suffer of memory leak when using the events emitted from
+        // TextField 2.0` implementation.
```

Comment:
But you can just pass the project_name as a weak reference to the closure and avoid leaking memory, cannot you?